usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedAgency.java

```diff
@@ -0,0 +1,88 @@
+ */
+public class ProcessParsedAgency {
+    private final ValidationResultRepository resultRepository;
+    private final GtfsDataRepository gtfsDataRepository;
+    private final Agency.AgencyBuilder builder;
+
+
+    public ProcessParsedAgency(final ValidationResultRepository resultRepository,
+                               final GtfsDataRepository gtfsDataRepository,
+                               final Agency.AgencyBuilder builder) {
+        this.resultRepository = resultRepository;
+        this.gtfsDataRepository = gtfsDataRepository;
+        this.builder = builder;
+    }
+
+    /**
+     * Use case execution method to go from a row from agency.txt to an internal representation.
+     * <p>
+     * This use case extracts values from a {@code ParsedEntity} and creates a {@code Agency} object if the requirements
+     * from the official GTFS specification are met. When these requirements are not met, related notices generated in
+     * {@code Agency.AgencyBuilder} are added to the result repository provided in the constructor.
+     * This use case also adds a {@code EntityMustBeUniqueNotice} to said repository if the uniqueness constraint on
+     * agency entities is not respected.
+     *
+     * @param validatedAgencyEntity entity to be processed and added to the GTFS data repository
+     */
+    public void execute(final ParsedEntity validatedAgencyEntity, final List<Notice> noticeCollection) {
+
+        final String agencyId = (String) validatedAgencyEntity.get("agency_id");
+        final String agencyName = (String) validatedAgencyEntity.get("agency_name");
+        final String agencyUrl = (String) validatedAgencyEntity.get("agency_url");
+        final String agencyTimezone = (String) validatedAgencyEntity.get("agency_timezone");
+        final String agencyLang = (String) validatedAgencyEntity.get("agency_lang");
+        final String agencyPhone = (String) validatedAgencyEntity.get("agency_phone");
+        final String agencyFareUrl = (String) validatedAgencyEntity.get("agency_fare_url");
+        final String agencyEmail = (String) validatedAgencyEntity.get("agency_email");
+
+        builder.agencyId(agencyId)
+                .agencyName(agencyName)
+                .agencyUrl(agencyUrl)
+                .agencyTimezone(agencyTimezone)
+                .agencyLang(agencyLang)
+                .agencyPhone(agencyPhone)
+                .agencyFareUrl(agencyFareUrl)
+                .agencyEmail(agencyEmail);
+
+        final var agency = builder.build(noticeCollection);
+
+        if (agency.getState()) {
+            if (gtfsDataRepository.addAgency((Agency) agency.getData()) == null) {
+                resultRepository.addNotice(new EntityMustBeUniqueNotice("agency.txt", "agency_id",
+                        validatedAgencyEntity.getEntityId()));
+            }
+        } else {
+            //noinspection unchecked
+            ((List<Notice>) agency.getData()).forEach(resultRepository::addNotice);
```

Comment:
Encapsulate type casting

Reply:
After discussion, we decided not to encapsulate type casting for the moment since it would require to cover every the possibility of all Gtfs entities, which raised the following question: should we have a general type for all gtfs entities?